zilliqa/src/consensus.rs

```diff
@@ -302,27 +298,63 @@ impl Consensus {
     }
 
     fn download_blocks_up_to(&mut self, to: u64) -> Result<()> {
+
         for view in (self.view + 1)..to {
             self.view = view;
             self.block_store.request_block_by_view(view)?;
         }
-        self.view = to + 1;
+
+        if to > self.view {
+            info!(
+                "*** Downloading missing blocks...view is going from {} to {}",
+                self.view,
+                to + 1
+            );
+            self.view = to + 1;
+        }
 
         Ok(())
     }
 
-    pub fn timeout(&mut self) -> Result<(PeerId, NewView)> {
+    pub fn timeout(&mut self, peers_joined: bool) -> Option<((PeerId, ExternalMessage))> {
+
+        if self.view == 1  {
```

Comment:
It doesn't make sense to increment the view past 1, so either do nothing, or if a peer has just joined, create a vote for genesis (starting the chain effectively)